core/storage/improvements/gae_models.py

```diff
@@ -0,0 +1,265 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models related to Oppia improvement tasks."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+import feconf
+
+from google.appengine.ext import ndb
+
+(base_models,) = models.Registry.import_models([models.NAMES.base_model])
+
+TEST_ONLY_ENTITY_TYPE = 'TEST_ONLY_ENTITY_TYPE'
```

Comment:
There's no way around this because of our usage of `choices` in `ndb.StringProperty`. GAE rejects any other value, and I'd have to mock out the internals of the engine to bypass it (which, to me, sounded worse than using these test-only constants).

Reply:
But why use them at all? Just use an existing task/entity type. That's better than changing the (rather fundamental) definition of what's valid and what isn't.